src/AppInstallerRepositoryCore/Public/winget/ARPCorrelation.h

```diff
@@ -42,16 +42,65 @@ namespace AppInstaller::Repository::Correlation
         bool IsNewOrUpdated;
     };
 
-    struct ARPCorrelationResult
+    // One of the possible options that could be chosen for correlation.
+    struct CorrelationMeasure
+    {
+        // The value that the correlation algorithm assigned to the match with the package.
+        double Measure{};
+
+        // The package that was measured.
+        std::shared_ptr<AppInstaller::Repository::IPackageVersion> Package{};
+    };
+
+    // The result of a heuristics correlation attempt.
+    struct ARPHeuristicsCorrelationResult
     {
         // Correlated package from ARP
         std::shared_ptr<AppInstaller::Repository::IPackageVersion> Package{};
+
+        // The reason for the correlation (for diagnostics).
+        std::string Reason;
+
+        // The correlation metrics and their associated ARP package information (for diagnostics).
+        std::vector<CorrelationMeasure> Measures;
+    };
+
+    // The result of a correlation attempt.
+    struct ARPCorrelationResult : public ARPHeuristicsCorrelationResult
+    {
         // Number of ARP entries that are new or updated
         size_t ChangesToARP{};
+
         // Number of ARP entries that match with the installed package
         size_t MatchesInARP{};
+
         // Number of changed ARP entries that match the installed package
         size_t CountOfIntersectionOfChangesAndMatches{};
+
+        ARPCorrelationResult& operator=(ARPHeuristicsCorrelationResult&& other)

```

Comment:
>ARPCorrelationResult& operator=(ARPHeuristicsCorrelationResult&& other) [](http://example.com/codeflow?start=8&length=71)

will this be misused?

Reply:
Honestly, I was surprised it didn't work out of the box.  The goal is to be able to assign just the base portion of the object, which seems very intuitive to me.